### PR TITLE
TF-1865: Modify sending email in queue 

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #1865 

- Resolved: 

1. Modify sending email in composer

[device-2023-06-05-002116.webm](https://github.com/linagora/tmail-flutter/assets/99852347/7f7b8e12-928d-454b-aa35-2a7187279ed8)

2. Replace task in workmanager

https://github.com/linagora/tmail-flutter/assets/99852347/f570b7a4-faf1-49c1-b9c6-da7fd9677644

3. Replace multi tasks in workmanager

https://github.com/linagora/tmail-flutter/assets/99852347/c2a498f5-a5dc-44c8-8c84-3d21eaf81310

### CODE REVIEW

**Comment by hoangdat** [comment]
How about when user `tap` in an item in sending queue?

File: assets/images/ic_resend_email.svg
Comment (hoangdat) [inline assets/images/ic_resend_email.svg:1]: please find it in `Icons.xxxx` first. We are going with material icon.

File: lib/features/sending_queue/presentation/sending_queue_controller.dart [hunk 1]
```diff
@@ -9,6 +18,12 @@ class SendingQueueController extends BaseController {
   final dashboardController = getBinding<MailboxDashBoardController>();
 
   final listSendingEmailController = ScrollController();
+  final FocusNode focusNodeKeyBoard = FocusNode();
```
Comment (dab246) [inline lib/features/sending_queue/presentation/sending_queue_controller.dart:21]: why need FocusNode at here?

File: lib/features/sending_queue/presentation/sending_queue_view.dart [hunk 2]
```diff
@@ -23,31 +31,86 @@ class SendingQueueView extends GetWidget<SendingQueueController> with AppLoaderM
...
+        );
+      } else {
+        return const SizedBox.shrink();
+      }
+    });
+  }
+
+  Widget _buildListButtonSelectionForMobile(BuildContext context) {
+    return Obx(() {
+      if ((!BuildUtils.isWeb || (BuildUtils.isWeb && controller.isSelectionEnabled()
```
Comment (dab246) [inline lib/features/sending_queue/presentation/sending_queue_view.dart:102]: SendingQueueView only displayed on` mobile` so you don't need to check `isWeb` condition

File: model/lib/email/email_action_type.dart [hunk 3]
```diff
@@ -17,5 +17,6 @@ enum EmailActionType {
   selection,
   moveToSpam,
   unSpam,
-  openInNewTab
+  openInNewTab,
+  resend
```
Comment (dab246) [inline model/lib/email/email_action_type.dart:21]: `resend`: this is not an action for `email`, it is only an action for `sending_email`

**Review by dab246 — COMMENTED** [review]
Screenshot 2023-06-05 at 11 49 52

Just do the `Edit`, Remove the other buttons.

**Review by dab246 — COMMENTED** [review]
Screenshot 2023-06-05 at 11 49 52

Just do the `Edit`, Remove the other buttons.

File: lib/features/email/domain/repository/email_repository.dart [hunk 4]
```diff
@@ -102,7 +102,7 @@ abstract class EmailRepository {
 
   Future<DetailedEmail?> getOpenedEmail(Session session, AccountId accountId, EmailId emailId);
 
-  Future<void> storeSendingEmail(AccountId accountId, UserName userName, SendingEmail sendingEmail);
+  Future<void> storeSendingEmail(AccountId accountId, UserName userName, SendingEmail sendingEmail, bool isUpdateSendingEmail);
```
Comment (dab246) [inline lib/features/email/domain/repository/email_repository.dart:105]: Should be separated into 2 different functions for `store` and `update`

File: model/lib/extensions/email_extension.dart [hunk 5]
```diff
@@ -135,4 +136,33 @@ extension EmailExtension on Email {
     }
     return null;
   }
+
+  PresentationEmail sendingEmailToPresentationEmail(
+    {
+      SelectMode selectMode = SelectMode.INACTIVE,
+      EmailId? emailId,
+      UTCDate? createAt,
```
Comment (dab246) [inline model/lib/extensions/email_extension.dart:144]: createAt of `email` is dynamically generated, you don't need to change it.

File: lib/features/composer/domain/extensions/list_sending_email_extension.dart [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/composer/domain/model/sending_email.dart';
+
+extension SendingEmailExtension on List<SendingEmail> {
+  List<PresentationEmail> toPresentationEmailList() {
+    return map((sendingEmail) => sendingEmail.email.sendingEmailToPresentationEmail(
+      emailId: EmailId(Id(sendingEmail.sendingId)),
```
Comment (dab246) [inline lib/features/composer/domain/extensions/list_sending_email_extension.dart:9]: `EmailId` is generated automatically from the system, we do not need to initialize it.

**Comment by hoangdat** [comment]
Some UX rules:
- Only edit/delete in offline mode 
- Can not edit/delete when connection comeback: Because at that time we can not manage the tasks
- Only Resend with the error item in queue

All were updated in https://www.figma.com/file/rRVxJpCu1G22D0jqyvK7I9/TeamMail?type=design&node-id=11122-431714&t=z3moVdufbdCMjPUf-4

**Comment by hoangdat** [comment]
We should split the ticket: Edit/Delete/Resend in separated ticket @dab246 @nqhhdev

**Comment by dab246** [comment]
- Only `Edit` when no network

**Review by dab246 — APPROVED** [review]

**Pull request merged**